sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -304,33 +308,78 @@ export class ManagementClient extends LinkEntity {
   private async _makeManagementRequest(
     request: Message,
     options?: { retryOptions?: RetryOptions; abortSignal?: AbortSignalLike; requestName?: string }
-  ): Promise<any> {
+  ): Promise<any> { 
     try {
-      log.mgmt(
-        "[%s] Acquiring lock to get the management req res link.",
-        this._context.connectionId
-      );
-      await defaultLock.acquire(this.managementLock, () => {
-        return this._init();
-      });
-
       if (!options) {
         options = {};
       }
 
-      const sendRequestOptions: SendRequestOptions = {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+
+          let waitTimer: any;
+          const actionAfterTimeout = () => {
+            const address = this._mgmtReqResLink!.receiver.address || "address";
+            const desc: string =
+              `The request with message_id "${request.message_id}" to "${address}" ` +
+              `endpoint timed out. Please try again later.`;
+            const e: Error = {
+              name: "OperationTimeoutError",
+              message: desc
+            };
+            return reject(translate(e));
+          };
+
+          waitTimer = setTimeout(
+            actionAfterTimeout,
+            getRetryAttemptTimeoutInMs(options!.retryOptions)
+          );
+
+          const initOperationStartTime = Date.now();
+          log.mgmt(
+            "[%s] Acquiring lock to get the management req res link.",
+            this._context.connectionId
+          );
+          await defaultLock.acquire(this.managementLock, () => {
+            return this._init().catch((err: Error) => {
+              clearTimeout(waitTimer);
+              reject(translate(err));
+            });
+          });
+          const initOperationEndTime = Date.now();
+
```

Comment:
You could check the abortSignal before calling the management API since it's possible the customer has cancelled the operation while the link was being created.

Reply:
I had thought of this, made changes and reverted back. My reeasoning to not have this was because link creation initialization isn't the actual operation. And it happens as part of `onDetached` mechanism as well. So technically it could keep failing, getting re-established outside of send as well, and thus, it seems irrelevant. 
And that overall it's sufficient that the actual send operation is cancelled when user aborts.

@ramya-rao-a thoughts? Also, this had involved quite some amount of refactoring. I was concerned you guys wanted to keep the changes minimal and specific to just removing the default retries from `core-amqp` :)

Reply:
One of the intentions of the cancellation token passed to a user facing function is that when it is fired, no matter what is happening inside the function, we try our best to return the control to the user.

Therefore, after the init() and before registering of handlers is a good place to check the abort signal.

What happens in `onDetached()` does not affect the flow of control for the user facing api.
Also, side note: This is the mangement link, we do not have the `onDetached()` for it.

Reply:
Got it - about
> after the init() and before registering of handlers is a good place to check the abort signal.

This is what the code is currently doing unless we meant _before_ the init()

Reply:
If we assume that `this._mgmtReqResLink!.sendRequest()` will check the abort signal first before doing anything else, then we don't have to add the same check here.

Reply:
Updating the thread here on the expectation of using the abort signal
- Before starting the timer, check for `aborted` property so that we can do an early exit before starting any async work
- Attach event listener for the `abort` event before calling `init()`. This is so that we can do an exit as soon as the signal gets fired instead of waiting for `init()` to succeed/fail
- When `init()` succeeds or fails, remove the event listener for `abort` event. This is because the next step is to call `sendRequest` which will take care of adding its own event listener for `abort`
- A second check on `aborted` property after `init()` completes and before `sendRequest` gets called is not needed. This is because the `sendRequest()` internally is expected to do the same check and do an early exit if needed

Reply:
I see the check for early exit based on `aborted` property and the adding of event listener for `abort` event before calling `init()`. This is good, but we still don't have the 3rd point in the previous comment covered which is as below:

> When init() succeeds or fails, remove the event listener for abort event. 

Also, since the `sendRequest()` method has its own timer and abort signal handling, it makes sense to pull the outer timer and abort signal handling into the `if(!this._isMgmtRequestResponseLinkOpen())` block. Please consider the following which simplifies the code

```typescript
const retryTimeoutInMs = getRetryAttemptTimeoutInMs(options.retryOptions);
let timeTakenByInit = 0;
if (!this._isMgmtRequestResponseLinkOpen()) {
     // TODO: Define rejectOnAbort and onAbort
     // TODO: early exit if aborted is true
     // TODO: add event listener for abort event
     const initOperationStartTime = Date.now();
     // TODO: start timer using retryTimeoutInMs constant
      try {
          await defaultLock.acquire(this.managementLock, this._init);
      } catch (err) {
           // TODO: clear timer and remove listener for abort event
          return reject(translate(err));
       }
      // TODO: clear timer and remove listener for abort event
       timeTakenByInit  = Date.now() - initOperationStartTime;
}
const remainingOperationTimeoutInMs = retryTimeoutInMs - timeTakenByInit;
// TODO: Call `sendRequest` using `retry()`
```

Reply:
Updated